src/components/learner-credit-management/BudgetAssignmentsTable.jsx

```diff
@@ -13,46 +13,138 @@ const BudgetAssignmentsTable = ({
+    <Button onClick={() => console.log('Remind', selectedFlatRows)}>
+      {`Remind (${selectedFlatRows.length})`}
+    </Button>
+  );
+  
+  const ExtraAction = ({ text, selectedFlatRows, ...rest }) => (
+    <Button variant="outline-primary" onClick={() => console.log(`Extra Action ${text}`, selectedFlatRows, rest)}>
+      {`Refresh`}
+    </Button>
+  );
+
+  const rowSelected = (rowsSelected) => {
+    if (Object.keys(rowsSelected).length > 1) {
+      setRowsSelected(true);
+    } else {
+      setRowsSelected(false);
+    }
+  };
+
+  const renderBulkActions = (selectedFlatRows) => {
+    if (rowsSelected) {
+      return ([
+        <RemindAction />,
+        <CancelAction />,
+        <ExtraAction />,
+      ]);
+    } else {
+      return ([]);
+    }
+  }
+  return (
+    <DataTable
+      isSortable
+      isSelectable
+      manualSortBy
+      isPaginated
+      manualPagination
+      isFilterable
+      manualFilters
+      isLoading={isLoading}
+      defaultColumnValues={{ Filter: TableTextFilter }}
+      FilterStatusComponent={FilterStatus}
+      columns={[
+        {
+          Header: 'Assignment details',
+          accessor: 'assignmentDetails',
+          Cell: AssignmentDetailsTableCell,
+          disableSortBy: true,
+        },
+        {
+          Header: 'Amount',
+          Cell: ({ row }) => `-${formatPrice(row.original.contentQuantity / 100, { maximumFractionDigits: 0 })}`,
+          disableFilters: true,
+        },
+      ]}
+      additionalColumns={[
+        {
+          id: 'action',
+          Header: '',
+          Cell: ({ row }) => (
+            <>
+              {!rowsSelected && (
+                <div className="d-flex">
+                  {row.original.state === 'allocated' && (
+                  <OverlayTrigger
+                    key="Remind"
+                    placement="top"
+                    overlay={
+                      <Tooltip variant="light" id={`tooltip-remind`}>
+                        Remind learner
+                      </Tooltip>
+                    }
+                  >
+                    <Icon
+                      className="ml-auto mr-0"
+                      src={Mail}
+                      onClick={() => console.log(`Reminding ${row.allCells}`)}
+                    />
+                  </OverlayTrigger>
+                )}
+                <div />
+                  <OverlayTrigger
+                    key="Cancel"
+                    placement="top"
+                    overlay={
+                      <Tooltip variant="light" id={`tooltip-cancel`}>
+                        Cancel assignment
+                      </Tooltip>
+                    }
+                  >
+                    <Icon
+                      className="ml-4.5 mr-2 text-danger-500"
+                      src={DoNotDisturbOn}
+                      onClick={() => console.log(`Cancelling ${Object.values(row.allCells)}`)}
```

Comment:
Ok this is a nit but just so we don't get confused later, this should be Canceling because of this annoying weirdness  https://www.scribbr.com/us-vs-uk/cancelled-or-canceled/#:~:text=Cancelling%20or%20canceling,-The%20same%20distinction&text=In%20UK%20English%2C%20%E2%80%9Ccancelling%E2%80%9D,is%20the%20more%20common%20spelling.

Reply:
oh lol, sure